### ISSUE TITLE
Watchers miss documents which have not yet made it to Elasticsearch

### ISSUE THREAD

**Issue by ypid-geberit** [elastic/examples#221]
Example from [errors_in_logs](https://github.com/elastic/examples/blob/master/Alerting/Sample%20Watches/errors_in_logs/watch.json):

```JSON
"range": {
  "@timestamp": {
    "gte": "now-1m"
  }
}
```

What happens to logs where the timestamp is in the range of `now` and `now-1m` which take a few seconds to process in the log collection pipeline? Those would be missed out which is not desirable. I would suggest to change the examples to something like:

```JSON
"range": {
  "@timestamp": {
    "gte": "now-6m",
    "lt": "now-5m"
  }
}
```

So we leave 5 minutes time for logs to reach Elasticsearch. Also note that I added a `lt` to prevent logs where the log source systems clock is in the future (happens) to be included multiple times.

Related to: https://discuss.elastic.co/t/ensure-that-watcher-does-not-miss-documents-logs/127780/1

PS: I can not yet contribute to Elastic in my work hours because the CLA is not yet signed by my company. Need to push them a bit more. The rebasing of the watcher integrations in this repo sucks … But that is not the fault of Elastic.

**Comment by cdahlqvist** [comment]
I agree specifying an interval in the past is a good practice in order to catch all records, and we should probably change the example to cover this and also discuss this a bit more. For a general example I would however not assume the user has such a large lag in their ingest pipeline, so picking an interval one minute in the past might be a good compromise:

```
"range": {
  "@timestamp": {
    "gte": "now-2m",
    "lt": "now-1m"
  }
}
```
Another way to get this more accurate is to add a timestamp that represents the ingestion time, and base the selection criteria on this rather than the the timestamp the event originated.
